engine/flink/components/table/src/main/scala/pl/touk/nussknacker/engine/flink/table/utils/BestEffortTableTypeSchemaEncoder.scala

```diff
@@ -0,0 +1,131 @@
+package pl.touk.nussknacker.engine.flink.table.utils
+
+import org.apache.flink.table.types.logical._
+import org.apache.flink.types.Row
+import org.springframework.util.NumberUtils
+import pl.touk.nussknacker.engine.api.typed.typing._
+import pl.touk.nussknacker.engine.flink.api.TypedMultiset
+import pl.touk.nussknacker.engine.flink.table.utils.DataTypesExtensions._
+
+import java.util
+import scala.jdk.CollectionConverters._
+import scala.reflect.ClassTag
+
+object BestEffortTableTypeSchemaEncoder {
```

Comment:
maybe `BestEffortToTableTypeEncoder`?

Reply:
This name I used in the next PR, I call these two encoders the same as in json case: `BestEffortJsonSchemaEncoder` and `BestEffortJsonEncoder`. But we can figure out some better names and rename all BestEffor*Encoders